### PR TITLE
Settings Refinements

### PR DESCRIPTION
**Pull request by Sjmarf**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [ ] This PR addresses one or more open issues that were assigned to me:
      - *list issue(s) here*
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

Changes listed below; let me know what you think.

- Added iOS-Settings-style labels to navigation links - #50.
- Moved some settings around:
    - Moved "Clear Cache" setting into a new 'advanced' menu. The average user won't care about this setting, so it shouldn't go under General in my opinion.
    - Moved "Blur NSFW" from Appearance Settings to General Settings - I think it fits better here.
    - Moved "App Icon" setting from the root settings page to Appearance Settings.
    - Categorised the toggles that were previously under Appearance Settings into four subsections - Posts, Comments, Communities and Users. I have just thrown the settings into these pages without much thought; in upcoming PRs I will be revisiting these sections as discussed in the Matrix room.
    - Moved the 'Documents' view that was previously under the root settings view into "About Mlem", where it has been split into three sections - Privacy Policy, EULA, and licenses. 
    - Moved the Contributors list that was previously under About Mlem to a "Contributors" subview of About Mlem.
- Added links to the Mlem Website, Matrix Room, Git Repo and Mlem Community under About Mlem. The link to the Mlem Community opens in-app. These were previously under the root settings view, but were commented out. The icon used for the Git Repo was sourced from [here](https://github.com/cameronshemilt/MoreSFSymbols/tree/main).

# Images

### CODE REVIEW

**Comment by WestonHanners** [comment]
@Sjmarf can you please re-target this to `dev` branch? master is currently in submission to the AppStore.

**Comment by Sjmarf** [comment]
> @Sjmarf can you please re-target this to `dev` branch? master is currently in submission to the AppStore.

Done 👍

**Comment by EricBAndrews** [comment]
That looks great!

Let's tweak the colors on that main page a bit, though--the two blues and the brown offer a pretty unappealing look. Let's make appearance pink and advanced gray--that should cheer it up a little

**Comment by Sjmarf** [comment]
> That looks great!
> 
> Let's tweak the colors on that main page a bit, though--the two blues and the brown offer a pretty unappealing look. Let's make appearance pink and advanced gray--that should cheer it up a little

Sure. Here's what it looks like now:

File: Mlem/Views/Tabs/Settings/Components/Views/About/AboutView.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                Section {
+                    Link(destination: URL(string: "https://mlem.group/")!) {
+                        Label("Website", systemImage: "globe").labelStyle(SquircleLabelStyle(color: .blue))
+                    }
+                    .buttonStyle(.plain)
+                    
+                    Button {
+                        Task {
+                            do {
+                                let request = GetCommunityRequest(account: appState.currentActiveAccount, communityId: 346)
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/About/AboutView.swift:32]: Communities have different IDs depending on which instance the user is logged into--it's 346 on sh.itjust.works, but 81081 on lemmy.ml

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    NavigationLink {
+                        IconSettingsView()
+                    } label: {
+                        Label {
+                           Text("App Icon")
+                        } icon: {
+                            IconSettingsView.getCurrentIcon()
+                                .resizable()
+                                .scaledToFit()
+                                .frame(width: 28, height: 28)
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift:42]: This 28 value gets reused a bunch--it should live in AppConstants as `settingsIconSize` or something like that

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                                .cornerRadius(7)
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift:43]: For consistency with the rest of the app, this should either be `AppConstants.smallItemCornerRadius` or `AppConstants.largeItemCornerRadius`--let's try it with small, if that looks bad/too blocky we can go with large

File: Mlem/Views/Tabs/Settings/Components/Views/About/AboutView.swift [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        if let buildVersion = Bundle.main.buildVersionNumber {
+            result.append(" (\(buildVersion))")
+        }
+
+        return result
+    }
+    
+    @ViewBuilder
+    private var appHeaderView: some View {
+        VStack(spacing: 10) {
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/About/AboutView.swift:108]: 10 should be `AppConstants.postAndCommentSpacing` (that's our default spacing--the name is long overdue for a refactor lol)

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/Comment/CommentSettingsView.swift [hunk 5]
```diff
@@ -8,23 +8,38 @@
 import Foundation
 import SwiftUI
 
-struct CustomizeCommentView: View {
+struct CommentSettingsView: View {
+    
+    @AppStorage("voteComplexOnRight") var shouldShowVoteComplexOnRight: Bool = false
+    @AppStorage("thumbnailsOnRight") var shouldShowThumbnailsOnRight: Bool = false
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/Appearance/Comment/CommentSettingsView.swift:14]: Is this value used in this file?

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/Theme/ThemeSettingsView.swift [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                        RoundedRectangle(cornerRadius: 7)
+                            .stroke(outlineColor, lineWidth: 1)
+                    }
+                
+            } else {
+                color1
+                    .frame(width: 28, height: 28)
+                    .clipShape(RoundedRectangle(cornerRadius: 7))
+                    .overlay {
+                        RoundedRectangle(cornerRadius: 7)
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Settings/Components/Views/Appearance/Theme/ThemeSettingsView.swift:48]: Ditto earlier comments about re: `AppConstants.settingsIconSize` and `AppConstants.smallItemCornerRadius`

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Needs a couple small changes and a rebase/merge with dev, but otherwise looks good!

**Comment by Sjmarf** [comment]
> Couple of small things, however when I pulled the branch I was unable to build it as there appears to be some issues with the file references for the alternative icons views too

Hmm interesting. It builds fine for me (on the latest beta). I could try removing and re-adding all of the files maybe?

File: Mlem/Views/Tabs/Settings/Components/Settings Item.swift [hunk 7]
```diff
@@ -47,3 +47,49 @@ struct SelectableSettingsItem<T: SettingsOptions>: View {
         }
     }
 }
+
+struct SquircleLabelStyle: LabelStyle {
+    var color: Color
+    var fontSize: CGFloat = 17
```
[the remark this answers is not in the document]
Comment (Sjmarf) [reply Mlem/Views/Tabs/Settings/Components/Settings Item.swift:53]: The font size refers to the icon font size and doesn't apply to the text on the label.

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift (hunk 3 above)
Comment (Sjmarf) [reply Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift:43]: This should be its own constant imo - smallItemCornerRadius is used for the Headline/Compact post thumbnails, which doesn't relate to the settings labels.

Comment (Sjmarf) [reply Mlem/Views/Tabs/Settings/Components/Views/Appearance/AppearanceSettingsView.swift]: I've added a new `AppConstants.settingsIconCornerRadius` value. What do you think?

File: Mlem/Views/Tabs/Settings/Components/Views/Appearance/Comment/CommentSettingsView.swift (hunk 5 above)
Comment (Sjmarf) [reply Mlem/Views/Tabs/Settings/Components/Views/Appearance/Comment/CommentSettingsView.swift]: No it isn't. I'll remove it 👍

**Comment by Sjmarf** [comment]
Resolved conflicts (I think).

**Comment by EricBAndrews** [comment]
I genuinely hate to do this to you, but the git log on this PR has become beyond broken--all the changes since it was opened have gotten incorporated into it, so the history is very hard to accurately parse and it's going to play merry hell with the main log.

Normally I'd suggest a rebase, but I think that would grab the already committed commits that have gotten tangled up in this one and just try to replay those, so, unless there's git wizardry I'm missing, we've got to do it manually:

- Open a new branch based on the current state of `dev`
- In order, cherry-pick *just your* commits:
  - 81f6821162c9543f449daa837ad3b571bb670f9b
  - a5cee211ff2f87b233cc805d788643f168b0370f
  - 6590aa29bcbfb48484641bbb7222b540575883e4
  - 90fd9a43514e108f4e78493a76bc1385c7f064a4
  - 5003607642cbfc19c897fa6345cfbb311b590eff
  - 82f50b2174e95904994b6dff06817704a93a0c7a
  - e7ce25b9956fdca0ec0e222d47027def0989d39a
  - fcb16117de3d48c8e84824285a79bd4983f63385
  - f078eba873ecff66dfdda7803e90a1d131d95786
  - 659e42802f444b2ffc8ca1397dfcb44207aa2d4d
 - Reopen this PR with that branch, which should have a nice clean history

I'm freezing anything else that touches settings until this one is in--you've suffered enough :P

**Comment by Sjmarf** [comment]
> * Reopen this PR with that branch, which should have a nice clean history

Done, I think 😅
# 338

**Pull request closed without merging**